components/FlexibleHeader/src/MDCFlexibleHeaderView.m

```diff
@@ -433,29 +405,60 @@ - (void)safeAreaInsetsDidChange {
   if (@available(iOS 11.0, *)) {
     [super safeAreaInsetsDidChange];
 
-    if (self.inferTopSafeAreaInsetFromViewController) {
-      // Will eventually invoke fhv_topSafeAreaInsetDidChange if needed.
-      [self extractTopSafeAreaInset];
-
-    } else {
-      [self fhv_topSafeAreaInsetDidChange];
-    }
+    [_topSafeArea safeAreaInsetsDidChange];
   }
 }
 
 #pragma mark - Top Safe Area Inset
 
-- (void)fhv_topSafeAreaInsetDidChange {
+- (id)topSafeAreaGuide {
+  return _topSafeAreaGuide;
+}
+
+- (CGFloat)topSafeAreaGuideHeight {
+  return _topSafeAreaGuide.frame.size.height;
+}
+
+- (BOOL)trackingScrollViewIsWebKit {
+  return [self.trackingScrollView.superview.class isSubclassOfClass:_wkWebViewClass];
+}
+
+#pragma mark - MDCFlexibleHeaderSafeAreas
+
+- (void)setInferTopSafeAreaInsetFromViewController:(BOOL)inferTopSafeAreaInsetFromViewController {
+  _topSafeArea.inferTopSafeAreaInsetFromViewController = inferTopSafeAreaInsetFromViewController;
+}
+
+- (BOOL)inferTopSafeAreaInsetFromViewController {
+  return _topSafeArea.inferTopSafeAreaInsetFromViewController;
+}
+
+- (void)setTopSafeAreaSourceViewController:(UIViewController *)topSafeAreaSourceViewController {
+  _topSafeArea.topSafeAreaSourceViewController = topSafeAreaSourceViewController;
+}
+
+- (UIViewController *)topSafeAreaSourceViewController {
+  return _topSafeArea.topSafeAreaSourceViewController;
+}
+
+- (void)topSafeAreaInsetDidChange {
+  [_topSafeArea safeAreaInsetsDidChange];
+}
+
+#pragma mark MDCFlexibleHeaderSafeAreasDelegate
+
+- (void)flexibleHeaderSafeAreasTopSafeAreaInsetDidChange:(MDCFlexibleHeaderTopSafeArea *)safeAreas {
   // If the min or max height have been explicitly set, don't adjust anything if the values
   // already include a Safe Area inset.
   BOOL hasSetMinOrMaxHeight = _hasExplicitlySetMinHeight || _hasExplicitlySetMaxHeight;
   if (!hasSetMinOrMaxHeight && _minMaxHeightIncludesSafeArea) {
     // If we're using the defaults we need to update them to account for the new Safe Area inset.
-    _minimumHeight = kFlexibleHeaderDefaultHeight + [self fhv_topSafeAreaInset];
+    _minimumHeight = kFlexibleHeaderDefaultHeight + [_topSafeArea topSafeAreaInset];
```

Comment:
Same here, please consider a local variable for `[_topSafeArea topSafeAreaInset]`